src/states/node.rs

```diff
@@ -2708,66 +2662,38 @@ impl Node {
         result.with_value(true)
     }
 
-    fn send_section_split(&mut self, our_prefix: Prefix<XorName>, joining_node: XorName) {
+    fn send_section_split(&mut self, our_prefix: Prefix, joining_node: XorName) {
         for prefix in self.peer_mgr.routing_table().prefixes() {
-            let request_msg = RoutingMessage {
-                // this way of calculating the source avoids using the joining node as the route
-                src: Authority::Section(our_prefix.substituted_in(!joining_node)),
-                dst: Authority::PrefixSection(prefix),
-                content: MessageContent::SectionSplit(our_prefix, joining_node),
-            };
-            if let Err(err) = self.send_routing_message(request_msg) {
+            // this way of calculating the source avoids using the joining node as the route
+            let src = Authority::Section(our_prefix.substituted_in(!joining_node));
+            let dst = Authority::PrefixSection(prefix);
+            let content = MessageContent::SectionSplit(our_prefix, joining_node);
+            if let Err(err) = self.send_routing_message(src, dst, content) {
                 debug!("{:?} Failed to send SectionSplit: {:?}.", self, err);
             }
         }
     }
 
     fn merge_if_necessary(&mut self) {
-        if let Some(merge_details) = self.peer_mgr.should_merge() {
-            self.send_own_section_merge(merge_details);
-        }
-    }
-
-    fn send_own_section_merge(&mut self, merge_details: OwnMergeDetails<XorName>) {
-        let sections = merge_details.sections
-            .into_iter()
-            .map(|(prefix, members)| {
-                (prefix, self.peer_mgr.get_pub_ids(&members).into_iter().collect())
-            })
-            .collect();
-        let request_content = MessageContent::OwnSectionMerge {
-            sender_prefix: merge_details.sender_prefix,
-            merge_prefix: merge_details.merge_prefix,
-            sections: sections,
-        };
-        let src_name = self.peer_mgr.routing_table().our_prefix().lower_bound();
-        let request_msg = RoutingMessage {
-            src: Authority::Section(src_name),
-            dst: Authority::PrefixSection(merge_details.merge_prefix),
-            content: request_content.clone(),
-        };
-        if let Err(err) = self.send_routing_message(request_msg) {
-            debug!("{:?} Failed to send OwnSectionMerge: {:?}.", self, err);
+        if let Some((sender_prefix, merge_prefix, sections)) = self.peer_mgr.should_merge() {
+            let content = MessageContent::OwnSectionMerge(sections);
+            let src = Authority::PrefixSection(sender_prefix);
```

Comment:
Why is this now a `PrefixSection(sender_prefix)` rather than just a `Section`? We want to send the message just from our section right? Is it because that prefix only includes our section unless our section splits, which can't happen?

even though the PrefixSection of our section should only include our section, unless our section splits, right?).